packages/ui/test/working-groups/modals/AddWorkerStakeModal.test.tsx

```diff
@@ -0,0 +1,194 @@
+
+const useMyAccounts: { hasAccounts: boolean; allAccounts: Account[] } = {
+  hasAccounts: true,
+  allAccounts: [],
+}
+
+jest.mock('@/accounts/hooks/useMyAccounts', () => {
+  return {
+    useMyAccounts: () => useMyAccounts,
+  }
+})
+
+describe('UI: AddStakeModal', () => {
+  setupMockServer()
+  const api = stubApi()
+  let transfer: any
+
+  const mockModalContext = {
+    hideModal: jest.fn(),
+    modalData: {
+      worker: {
+        minStake: 300,
+        stake: 100,
+        roleAccount: alice.address,
+        id: 'workerId',
+        group: {
+          id: 'storageWorkingGroup',
+        },
+        runtimeId: 88,
+      },
+    },
+    modal: null,
+    showModal: jest.fn(),
+  }
+
+  beforeAll(async () => {
+    await cryptoWaitReady()
+    useMyAccounts.allAccounts.push(alice, bob)
+  })
+
+  beforeEach(async () => {
+    stubDefaultBalances(api)
+    transfer = stubTransaction(api, 'api.tx.storageWorkingGroup.increaseStake')
+  })
+
+  describe('Form step', () => {
+    it('Renders modal', () => {
+      const { queryByText } = renderModal()
+
+      expect(queryByText('Add stake')).toBeDefined()
+    })
+
+    it('Setup minimal stake', async () => {
+      renderModal()
+
+      const input = await getStakeInput()
+      const submitButton = await getButton('Add Stake')
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      expect(input?.value).toBe('200')
+      expect(submitButton).not.toBeDisabled()
+    })
+
+    it('Too low stake', async () => {
+      renderModal()
+
+      const input = await getStakeInput()
+      const submitButton = await getButton('Add Stake')
+
+      fireEvent.change(input, { target: { value: '100' } })
+
+      expect(submitButton).toBeDisabled()
+    })
+
+    it('Stake exceed balance', async () => {
+      renderModal()
+
+      const input = await getStakeInput()
+      const submitButton = await getButton('Add Stake')
+
+      fireEvent.change(input, { target: { value: '10000' } })
+
+      expect(submitButton).toBeDisabled()
+    })
+  })
+
+  describe('Sign step', () => {
+    it('Goes to sign modal', async () => {
+      const { queryByText } = await renderModalAndProceedToSign()
+
+      expect(queryByText('Authorize transaction')).toBeDefined()
+    })
+
+    it('Transaction exceed balance', async () => {
+      renderModal()
+
+      const input = await getStakeInput()
+      const submitButton = await getButton('Add Stake')
+
+      fireEvent.change(input, { target: { value: '1000' } })
+      submitButton.click()
+
+      const signButton = await getButton('Sign transaction and Stake')
+
+      expect(signButton).toBeDisabled()
+    })
+
+    describe('Pending', () => {
+      it('Renders wait for transaction', async () => {
+        stubTransactionPending(transfer)
+
+        await renderModalAndProceedToSign(true)
+
+        expect(screen.queryByText('Pending transaction')).toBeDefined()
+      })
+    })
```

Comment:
I think that pending step might be omitted. This is a general mechanism, so testing for success/error should be enough.